### ISSUE TITLE
Question: Why does the usePlaid hook require the config to be destructured to reinitialize?

### ISSUE THREAD

**Issue by sdzharkov** [plaid/react-plaid-link#93]
Hey Plaid team,

I have a quick question regarding the need to destructure the config that passed into the usePlaid hook. In your example, https://github.com/plaid/react-plaid-link/blob/master/src/PlaidLink.tsx#L8 you destructure and restructure the config object.

Initially, I tried to pass in an object into `usePlaid` like this: 
```js
const config_object = {
   ......
}
// Does not reinitialize Plaid when the config_object changes
const { error, open } = usePlaidLink(config_object)

// Reinitialized Plaid when the config_object changes
const { error, open } = usePlaidLink({ ...config_object})
 ```
but this would not reinitialize the plaid module. However, destructuring the config object works. Why is this the case?

Cheers,
Stephan

**Comment by abartolo** [comment]
@sdzharkov Any chance you came across and issue opening institutions after changing the plaid config? Every time I change the PlaidConfig and invoke `open` it doesn't open the institution.

**Comment by sdzharkov** [comment]
@abartolo I didn't have a problem with institutions, my problem was that the plaid link module would not reinitialize when I updated the `onSuccess` callback for the config object. The reason why it didn't reinitialize is because the hook's dependency array did not include `options.onSuccess`.

https://github.com/plaid/react-plaid-link/blob/e585aef061b18a69a00b456cc002407d1ff446bd/src/usePlaidLink.ts#L61

Potentially you might be experiencing the same thing on your end, where one of your config objects has been updated but the `usePlaidLink` hook is not reinitializing. Hope that helps!
